MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationComponent.java

```diff
@@ -1619,12 +1623,31 @@ private void updateLocation(@NonNull LocationUpdate locationUpdate, boolean from
     }
     CameraPosition currentCameraPosition = mapboxMap.getCameraPosition();
     boolean isGpsNorth = getCameraMode() == CameraMode.TRACKING_GPS_NORTH;
-    locationAnimatorCoordinator.feedNewLocation(
-      getTargetLocationWithIntermediates(locationUpdate.getLocation(), locationUpdate.getIntermediatePoints()),
-      currentCameraPosition,
-      isGpsNorth,
-      fromLastLocation ? Long.valueOf(0L) : locationUpdate.getAnimationDuration());
-    updateAccuracyRadius(locationUpdate.getLocation(), false);
+
+
+    //Performance improvement: Check if new location requires a redraw to be performed,
+    //taking the pixel tolerance into account.
+    //By default, the tolerance is 1 pixel, meaning that the new location must impact at least
+    //a different pixel on the screen to justify a redraw.
+    boolean updateNeeded = (lastLocation == null);
+    if (updateNeeded == false) {
+      PointF lastPoint = this.mapboxMap.getProjection().toScreenLocation(
+              new LatLng(lastLocation.getLatitude(), lastLocation.getLongitude()));
```

Comment:
Using projection on each frame adds up to CPU usage as well. We might want to cache this value the same as we do with `lastLocation` if we're comparing it each time anyway.

Reply:
Hi !
I did it and tested it, but it's not applicable.
As soon as the map moves (scroll, rotation, and so on) the PointF is deprecated and cannot be reused for comparaison.
This code is not run every frame, but only when a new location is provided.
Moreover, existing lastUpdateTime and fastestInterval already prevents this method to be called too fast.
So the projection cost stays very low.